src/compiler/factory.ts

```diff
@@ -2292,15 +2292,26 @@ namespace ts {
     /**
      * Sets a custom text range to use when emitting source maps.
      */
-    export function setSourceMapRange<T extends Node>(node: T, range: TextRange | undefined) {
+    export function setSourceMapRange<T extends Node>(node: T, range: SourceMapRange | undefined) {
         getOrCreateEmitNode(node).sourceMapRange = range;
         return node;
     }
 
     /**
+     * Create an external source map source file reference
+     */
+    export function createSourceMapSource(fileName: string, text: string, skipTrivia?: (pos: number) => number) {

```

Comment:
Out of curiosity, why must it extend `SourceFileLike`? Could you not use a union in the places where this matters in either `sourcemap.ts` or `utilities.ts`?